Common/Source/Dialogs/dlgLXIGCDownload.cpp

```diff
@@ -220,29 +220,28 @@ static void OnMultiSelectListPaintListItem(WindowControl *Sender, LKSurface &Sur
   if (LX_IGCReadDialog.FileList()->size() < (uint) LX_IGCReadDialog.DrawIndex()) return;
   if (LX_IGCReadDialog.DrawIndex() < LX_IGCReadDialog.FileList()->size()) {
     Surface.SetTextColor(RGB_BLACK);
+    TCHAR szTmp[MAX_NMEA_LEN] ;
     TCHAR text1[MAX_NMEA_LEN] = {TEXT("IGC File")};
     TCHAR text2[MAX_NMEA_LEN] = {TEXT("date")};
+    _tcscpy(szTmp, LX_IGCReadDialog.FileList()->at(LX_IGCReadDialog.DrawIndex()).Line1);
     _tcscpy(text1, LX_IGCReadDialog.FileList()->at(LX_IGCReadDialog.DrawIndex()).Line1);
     _tcscpy(text2, LX_IGCReadDialog.FileList()->at(LX_IGCReadDialog.DrawIndex()).Line2);
 
     TCHAR *remaining;    /* extract filname */
     TCHAR *IGCFilename = _tcstok_r(
-            LX_IGCReadDialog.FileList()->at(LX_IGCReadDialog.DrawIndex()).Line1, TEXT(" "),
+        szTmp, TEXT(" "),
             &remaining);
 
     TCHAR PathAndFilename[MAX_PATH];
     LocalPath(PathAndFilename, _T(LKD_LOGS), IGCFilename);     // add path
+    _sntprintf(text1, MAX_NMEA_LEN, _T("%s"),text1);     // missing
```

Comment:
result of using same buffer for source and destination is undefined

Reply:
done